src/components/Widgets/Markdown/serializers/remarkEscapeMarkdownEntities.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * A Remark plugin for escaping markdown entities.
+ *
+ * When markdown entities are entered in raw markdown, they don't appear as
+ * characters in the resulting AST; for example, dashes surrounding a piece of
+ * text cause the text to be inserted in a special node type, but the asterisks
+ * themselves aren't present as text. Therefore, we generally don't expect to
+ * encounter markdown characters in text nodes.
+ *
+ * However, the CMS visual editor does not interpret markdown characters, and
+ * users will expect these characters to be represented literally. In that case,
+ * we need to escape them, otherwise they'll be interpreted during
+ * stringification.
+ */
+export default function remarkEscapeMarkdownEntities() {
+  /**
+   * Escape all occurrences of '[', '*', '_', '`', and '~'.
+   */
+  function escapeCommonChars(text) {
+    return text.replace(/[\[*_`~]/g, '\\$&');
+  }
+
+  /**
+   * Runs escapeCommonChars, and also escapes '#' and '-' when found at the
+   * beginning of any node's first child node.
+   */
+  function escapeAllChars(text) {
+    const partiallyEscapedMarkdown = escapeCommonChars(text);
+    return partiallyEscapedMarkdown.replace(/^\s*([#-])/, '$`\\$1');
```

Comment:
This works fine, but I almost wonder if it would be simpler just to replace all characters all the time, just to avoid edge cases. Either way I'm fine with it.

Reply:
I'd prefer to leave the markdown as human-editable as possible and address edge cases as they occur.

Reply:
@tech4him1 keep in mind that any markdown entity in the raw editor that isn't active, e.g. a lone asterisk, will be escaped when the markdown is stringified. In other words, the user enters an asterisk in the raw editor, they save the entry, then they re-open and it's been escaped. To some extent it has to be that way, but we want to minimize occurrences of unexpected transformation.

This is an edge case, as Benaiah mentioned, that we want to tackle. These two characters are benign unless they are at the beginning of a line, so there's no need to escape them at any other time. As bug reports come on we'll probably be extending this plugin to handle more edge cases.